pkg/cmd/cli/restic/server.go

```diff
@@ -182,26 +203,27 @@ func (s *resticServer) run() {
 		s.secretInformer,
 		s.kubeInformerFactory.Core().V1().PersistentVolumeClaims(),
 		s.kubeInformerFactory.Core().V1().PersistentVolumes(),
-		s.veleroInformerFactory.Velero().V1().BackupStorageLocations(),
+		s.mgr.GetClient(),
 		os.Getenv("NODE_NAME"),
 	)
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		restoreController.Run(s.ctx, 1)
-	}()
 
 	go s.veleroInformerFactory.Start(s.ctx.Done())
 	go s.kubeInformerFactory.Start(s.ctx.Done())
 	go s.podInformer.Run(s.ctx.Done())
 	go s.secretInformer.Run(s.ctx.Done())
 
-	s.logger.Info("Controllers started successfully")
+	// Adding the controllers to the manager will register them as a (runtime-controller) runnable,
+	// so the manager will ensure the cache is started and ready before all controller are started
+	s.mgr.Add(managercontroller.Runnable(backupController, 1))
+	s.mgr.Add(managercontroller.Runnable(restoreController, 1))
 
-	<-s.ctx.Done()
+	s.logger.Info("Controllers starting...")
+
+	if err := s.mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+		s.logger.Fatal("Problem starting manager", err)
+	}
 
 	s.logger.Info("Waiting for all controllers to shut down gracefully")
```

Comment:
`s.mgr.Start` is a blocking call, so this message wouldn't be logged until `Start` returns.
I would suggest removing this log message.

Reply:
Good point!